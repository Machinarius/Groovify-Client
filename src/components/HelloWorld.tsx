import * as React from 'react';

export interface HelloProps {
    compiler: string,
    framework: string
}

export class HelloWorld extends React.Component<HelloProps, {}> {
    render() {
        return (<div>Hello from {this.props.compiler} and {this.props.framework}</div>);
    }
}
