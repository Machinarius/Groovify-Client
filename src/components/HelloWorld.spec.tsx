import * as React from 'react';
import * as testRenderer from 'react-test-renderer';

import { shallow } from 'enzyme';
import HelloWorld from './HelloWorld'

test("Must show the hello message for the prop values passed in", () => {
    let component = shallow(<HelloWorld compiler="TypeScript" framework="Jest" />);
    expect(component.text()).toBe("Hello from TypeScript and Jest");
});

test("Must match the snapshot", () => {
    let component = testRenderer.create(
        <HelloWorld compiler="TypeScript" framework="Jest" />
    );
    let treeData = component.toJSON();
    expect(treeData).toMatchSnapshot();
});