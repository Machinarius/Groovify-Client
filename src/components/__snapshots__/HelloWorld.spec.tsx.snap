// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Must match the snapshot 1`] = `
<div>
  Hello from 
  TypeScript
   and 
  Jest
</div>
`;
